engines/pytorch/pytorch-engine/src/main/java/ai/djl/pytorch/engine/PtGradientCollector.java

```diff
@@ -37,8 +39,18 @@ public PtGradientCollector() {
                     "A PtGradientCollector is already collecting. Only one can be collecting at a"
                             + " time");
         }
+    }
 
-        zeroGradients();
```

Comment:
I think we still want the zeroGradients here. Otherwise, it won't work correctly for cases without a block

Reply:
The reason of removing it was mainly due to efficiency and memory issue https://github.com/deepjavalibrary/djl/issues/2210.
Should we add this back and then do
```
    public GradientCollector newGradientCollector() {
        return manager.getEngine().newGradientCollector(getModel().getBlock());
    }
```
Does this avoid the global search scope issue?